src/impl_1d.rs

```diff
@@ -23,5 +26,77 @@ impl<A, S> ArrayBase<S, Ix1>
             ::iterators::to_vec(self.iter().map(|x| x.clone()))
         }
     }
+
+    /// Return the element that would occupy the `i`-th position if the array
+    /// were sorted in increasing order.
+    ///
+    /// The array is shuffled **in place** to retrieve the desired element:
+    /// no copy of the array is allocated.
+    /// No assumptions should be made on the ordering of elements
+    /// after this computation.
+    ///
+    /// Complexity ([quickselect](https://en.wikipedia.org/wiki/Quickselect)):
+    /// - average case: O(`n`);
+    /// - worst case: O(`n`^2);
+    /// where n is the number of elements in the array.
+    ///
+    /// **Panics** if `i` is greater than or equal to `n`.
+    pub fn ith_mut(&mut self, i: usize) -> A
+        where A: Ord + Clone,
+              S: DataMut,
+    {
+        let n = self.len();
+        if n == 1 {
+            (&self[0]).clone()
+        } else {
+            let pivot_index = random_pivot(n);
+            let partition_index = self.view_mut().partition_mut(pivot_index);
```

Comment:
I don't understand your usage of `view_mut()` here. Why can't you write

    let partition_index = self.partition_mut(pivot_index);

? We already know that `self` is mutable. Is it important if `self` is a `OwnedRepr` or a `ViewRepr`?

Reply:
It is actually unnecessary, given that we have the `S: DataMut` bound. I removed it.